app/objects/c_c2.py

```diff
@@ -0,0 +1,30 @@
+from app.utility.base_object import BaseObject
+
+
+class C2(BaseObject):
+    @property
+    def unique(self):
+        return '%s%s' % (self.module, self.config)
+
+    @property
+    def display(self):
+        return dict(module=self.module, config=self.config)
+
+    def __init__(self, c2_type=None, name='API', module=None, config={}, enabled=True):
+        self.name = name
+        self.module = module
+        self.config = config
+        self.c2_type = c2_type
+        self.enabled = enabled
+
+    def store(self, ram):
+        """
+        Store the object in ram
+        :param ram:
```

Comment:
does this object need to be a "first-class" object (meaning, it is intended to be stored and accessed like an 'agent' or an 'adversary'). or should it be more like a link, where it is a component of another object?

Reply:
We went back and forth on this, and settled (for the interim) on first class, with the intention being to make it component of agent objects in the long run.  This is the first iteration of the approach, with the idea being that everything will end up behind the C2_service as either a passive (API) or active (GIST, Slack, Drafts, etc) C2 channels.